protocol/rtsp/src/session/mod.rs

```diff
@@ -150,7 +151,12 @@ impl RtspServerSession {
                 match data {
                     Some(a) => {
                         if self.reader.len() < a.length as usize {
-                            let data = self.io.lock().await.read().await?;
+                            let data = self
+                                .io
+                                .lock()
+                                .await
+                                .read_min_bytes_with_timeout(Duration::from_millis(1000), a.length.into())
```

Comment:
the length parameter passed into the function maybe a.length - self.reader.len() ??

Reply:
ping @bailb